### DISCUSSION TITLE
osu! doesn't detect key presses sent by the win32api SendInput method

### DISCUSSION THREAD

**Discussion by creeperkafasi** [ppy/osu#15501] [Q&A]
I am working on a personal project that lets me send keyboard/mouse input from my phone to my pc over LAN. However for some reason osu! doesn't seem to detect the key presses I send on windows. It works fine in other programs and on linux. Is there some kind of anti-cheat that ignores the keypresses sent by win32api or does osu! somehow directly check the hardware keys on windows?

**Comment by bdach** [comment]
i am personally not aware of any such measures existing in lazer *yet*.

difficult to say more without debugging. you could use spy++ or other utility to see what window events are being emitted by your project, which may answer the question as to why they're not being handled.
